src/main/kotlin/no/nav/familie/ef/sak/beregning/BeregningUtils.kt

```diff
@@ -2,72 +2,99 @@ package no.nav.familie.ef.sak.beregning
+            val beløpFørSamordningUtenAvrunding =
+                    fullOvergangsStønadPerMåned.subtract(avkortningPerMåned).setScale(0, RoundingMode.HALF_UP)
 
-                val utbetaling = beløpFørSamordningUtenAvrunding.subtract(samordningsfradrag).setScale(0, RoundingMode.HALF_UP)
+            val utbetaling = beløpFørSamordningUtenAvrunding.subtract(samordningsfradrag).setScale(0, RoundingMode.HALF_UP)
 
-                val beløpFørSamordning =
-                        if (beløpFørSamordningUtenAvrunding <= BigDecimal.ZERO) BigDecimal.ZERO else beløpFørSamordningUtenAvrunding
+            val beløpFørSamordning =
+                    if (beløpFørSamordningUtenAvrunding <= BigDecimal.ZERO) BigDecimal.ZERO else beløpFørSamordningUtenAvrunding
 
-                val beløpTilUtbetalning = if (utbetaling <= BigDecimal.ZERO) BigDecimal.ZERO else utbetaling
+            val beløpTilUtbetalning = if (utbetaling <= BigDecimal.ZERO) BigDecimal.ZERO else utbetaling
 
-                Beløpsperiode(periode = it.periode,
-                              beløp = beløpTilUtbetalning,
-                              beløpFørSamordning = beløpFørSamordning,
-                              beregningsgrunnlag = Beregningsgrunnlag(samordningsfradrag = samordningsfradrag,
-                                                                      avkortningPerMåned = avkortningPerMåned,
-                                                                      fullOvergangsStønadPerMåned = fullOvergangsStønadPerMåned,
-                                                                      inntekt = inntekt,
-                                                                      grunnbeløp = it.beløp))
-            }
+            Beløpsperiode(periode = it.periode,
+                          beløp = beløpTilUtbetalning,
+                          beløpFørSamordning = beløpFørSamordning,
+                          beregningsgrunnlag = Beregningsgrunnlag(samordningsfradrag = samordningsfradrag,
+                                                                  avkortningPerMåned = avkortningPerMåned,
+                                                                  fullOvergangsStønadPerMåned = fullOvergangsStønadPerMåned,
+                                                                  inntekt = inntekt,
+                                                                  grunnbeløp = it.beløp))
         }
+    }
+
+    private fun beregnAvkortning(grunnbeløp: BigDecimal, inntekt: BigDecimal): BigDecimal {
+        val inntektOverHalveGrunnbeløp = inntekt.subtract(grunnbeløp.multiply(BigDecimal(0.5)))
+        return if (inntektOverHalveGrunnbeløp > BigDecimal.ZERO)
+            inntektOverHalveGrunnbeløp.multiply(REDUKSJONSFAKTOR).setScale(5, RoundingMode.HALF_DOWN) else BigDecimal.ZERO
+    }
+
 
+    fun indeksjusterInntekt(sisteGrunnbeløpsdato: LocalDate,
```

Comment:
Denne lurte meg litt - klarer vi å komme opp med et annet navn?
Skulle vi egentlig lagret ned dette som enum med datoverdi for å skjønne at det er en "fast" dato og ikke vilkårlig?

F.eks:
```
G_ORDINÆR_2020(startDatoForGrunnbeløpsPeriode=LocalDate.of(2020, 5, 1))
G_EKSTRAORDINÆR_2020(startDatoForGrunnbeløpsPeriode=LocalDate.of(2020, 7, 1))
```

Eller så lurer jeg på om et variabelnavn ala dette kan funke (eller noe bedre):
`GrunnbeløpsperiodeForSisteGjennomførteRegulering`

Reply:
Er det egentlig en "fullførtReguleringTilOgMedPeriodeSomOmslutterDato"?

Reply:
Oppdatert variabelnavn